pkg/auth/oauth2authenticator.go

```diff
@@ -193,6 +240,36 @@ func (o *oAuth2Authenticator) persistToken(token *oauth2.Token) {
 }
 
 func (o *oAuth2Authenticator) Authenticate() error {
+	var err error
+
+	if o.grantType == ClientCredentialsGrant {
+		err = o.authenticateWithClientCredentialsGrant()
+	} else {
+		err = o.authenticateWithAuthorizationCode()
+	}
+
+	return err
+}
+
+func (o *oAuth2Authenticator) authenticateWithClientCredentialsGrant() error {
+	ctx := context.Background()
+	config := getOAuthConfigurationClientCredentials(o.oauthConfig)
+
+	// Use the custom HTTP client when requesting a token.
+	if o.httpClient != nil {
+		ctx = context.WithValue(ctx, oauth2.HTTPClient, o.httpClient)
+	}
+
+	// get token
+	token, err := config.Token(ctx)
+	if err == nil {
```

Comment:
suggestion: I would invert that logic. It confuses me to have the default flow in the if, and the error flow outside.